### PR TITLE
Stick-Knife Trick

### PR DESCRIPTION
**Pull request by Ignose**

### CODE REVIEW

**Comment by Ignose** [comment]
Checks if you're PM, have the relevant skill, have stick-knife, and chooses an outfit based on that.

Outfit is tied to a hidden pref. If the pref should be not hidden, let me know. Tested in Folger today.

File: src/tasks/spelldamage.ts [hunk 1]
```diff
@@ -151,6 +155,20 @@ export const SpellDamageQuest: Quest = {
...
+        get("instant_stickKnifeOutfit") !== "" &&
+        myClass() === $class`Pastamancer` &&
+        have($item`Stick-Knife of Loathing`) &&
+        have($skill`Bind Undead Elbow Macaroni`),
+      completed: () => haveEquipped($item`Stick-Knife of Loathing`),
+      do: (): void => {
+        useSkill($skill`Bind Undead Elbow Macaroni`);
+        outfit(get("instant_stickKnifeOutfit"));
+      },
+      limit: { tries: 2 },
```
Comment (Pantocyclus) [inline src/tasks/spelldamage.ts:170]: What happens when you already have the Elbow Macaroni out as your active thrall?
1. Do you still `have($skill'Bind Undead Elbow Macaroni')`
2. Does `useSkill($skill'Bind Undead Elbow Macaroni')` not remove the active thrall?

Also, why does this need 2 tries? And will the maximizer (that is run in the test task) mess with the outfit?

Comment (Pantocyclus) [reply src/tasks/spelldamage.ts:170]: Seems like this task will fail if the user does not have at least 250 base myst, which may be the case for many low shiny users. 

On the other end of the spectrum, a less consequential point is that this seems like a waste of 100mp for users who already have a Disembodied Hand and thus do not need this trick.

Comment (Ignose) [reply src/tasks/spelldamage.ts:170]: I don't know. Does/would anything in the script make Elbow Macaroni your thrall?

This could be hardened for a That Guy, I suppose.

It doesn't need two tries; I'll drop to one.

Maximizer is mad about it, but it works:

Cast 1 Bind Undead Elbow Macaroni
Preference _concoctionDatabaseRefreshes changed from 348 to 349
custom outfit Stick-Knife
*** slot weapon: KoL has candy cane sword cane but KoLmafia has Stick-Knife of Loathing
*** slot off-hand: KoL has Stick-Knife of Loathing but KoLmafia has candy cane sword cane

Comment (Pantocyclus) [reply src/tasks/spelldamage.ts]: You're running this function in both wdmg and sdmg. If the thrall is not removed at the end of the first test, it probably would be present when running the task for the next test?

Comment (Ignose) [reply src/tasks/spelldamage.ts]: Yeah; I added an If to both to check for current thrall., and an Or current thrall to ready

I can add base mys and disembodied hand checks.

Comment (Ignose) [reply src/tasks/spelldamage.ts]: Also, it's 150 Mys. Which is below the 180 default threshhold.

File: src/tasks/spelldamage.ts [hunk 2]
```diff
@@ -151,6 +157,21 @@ export const SpellDamageQuest: Quest = {
...
+      ready: () =>
+        get("instant_stickKnifeOutfit") !== "" &&
+        myClass() === $class`Pastamancer` &&
+        have($item`Stick-Knife of Loathing`) &&
+        (have($skill`Bind Undead Elbow Macaroni`) || myThrall() === $thrall`Undead Elbow Macaroni`),
+      completed: () => haveEquipped($item`Stick-Knife of Loathing`),
+      do: (): void => {
+        if (myThrall() !== $thrall`Undead Elbow Macaroni`)
+          useSkill($skill`Bind Undead Elbow Macaroni`);
+        outfit(get("instant_stickKnifeOutfit"));
```
Comment (Pantocyclus) [inline src/tasks/spelldamage.ts:171]: I would add `|| have($familiar'Disembodied Hand') || my_baseStat($stat'Mysticality') < 250` to the completion condition, and do `if (myThrall() !== $thrall'Undead Elbow Macaroni' && my_baseStat($stat'Muscle') < 250)` to prior to summoning the thrall (there apparently are many high shiny users who appear to be hitting >= level 21 powerleveling, and they could very well have enough Muscle to equip the stick knife without the macaroni)

File: src/tasks/spelldamage.ts (hunk 1 above)
Comment (Pantocyclus) [reply src/tasks/spelldamage.ts]: Speaking of, how does cowrruption limiting base muscle to 30 interact with the thrall? I'm trying to think if there are any edge cases with base stat limiters

e.g. a Guy who pulls an equipment that was equipped for prior tests and so happens to limit base stats - if it limits base myst as well would that mess up this task? (This case doesn't mess up the equipping during the Test task since running the maximizer there should remove any such equipment)

Comment (Pantocyclus) [reply src/tasks/spelldamage.ts]: I was thinking that they could have pre-pulled it, and the maximizer for the previous test could possibly have equipped it. But I'm fine with leaving that for another day when the problem actually surfaces.

I'm still concerned about the Cowrruption though, and I do want to know for certain that this task would not break with the effect.

Comment (Ignose) [reply src/tasks/spelldamage.ts]: Used one in aftercore; I can still equip the stick-knife. It looks like  it's based on base stats.
